agent/metasploit_agent.py

```diff
@@ -92,52 +84,34 @@ def process(self, message: m.Message) -> None:
                     "Failed to set arguments for %s", selected_module.modulename
                 )
                 continue
-            if module_instance.moduletype == "exploit":
-                job = module_instance.check_exploit()
-            elif module_instance.moduletype == "auxiliary":
-                job = module_instance.execute()
-            else:
+            job = module_instance.check_exploit()
+            if job.get("error") is True:
                 logger.error(
-                    "%s module type is not implemented", module_instance.moduletype
+                    "Metasploit Error: %s", job.get("error_string", "Unknown Error")
                 )
                 continue
-            job_uuid = job.get("uuid")
-            if job_uuid is not None:
-                results = self._get_job_results(client, job_uuid)
-            else:
-                results = None
 
-            if isinstance(results, dict) and results.get("code") in ["safe", "unknown"]:
+            job_uuid = job.get("uuid")
+            if job_uuid is None:
                 continue
 
-            target = (
-                module_instance.runoptions.get("VHOST")
-                or module_instance.runoptions.get("RHOSTS")
-                or module_instance.runoptions.get("DOMAIN")
-            )
-            technical_detail = f"Using `{module_instance.moduletype}` module `{module_instance.modulename}`\n"
-            technical_detail += f"Target: {target}\n"
+            results = self._get_job_results(client, job_uuid)
 
-            if isinstance(results, dict) and results.get("code") == "vulnerable":
-                technical_detail += f'Message: \n```{results["message"]}```'
-            else:
-                console_output = client.consoles.console(cid).run_module_with_output(
-                    module_instance
+            if isinstance(results, dict) and results.get("code") in [
+                "vulnerable",
+                "appears",
+            ]:
+                target = (
```

Comment:
You can use the target class instead. I already added one in utils in my PR.

Reply:
No need as we only need a string of the target to show to the user